CalibCalorimetry/HGCalPlugins/plugins/HGCalConfigESSourceFromYAML.cc

```diff
@@ -88,33 +89,60 @@ class HGCalConfigESSourceFromYAML : public edm::ESProducer, public edm::EventSet
       cond->moduleConfigs[0] = HGCalModuleConfig();
       
-      // PARSE META DATA NODE
-      if (const auto config = yaml_file["metaData"]; config.IsDefined()) {
-        int charMode = yaml_file["metaData"]["characMode"].as<int>();
-        assert(charMode==0 or charMode==1);
-        cond->moduleConfigs[0].charMode = (bool) charMode;
-
-        cond->moduleConfigs[0].injcalib = 0;
-        if(yaml_file["metaData"]["chip_params"]["Calib"])
-          cond->moduleConfigs[0].injcalib = yaml_file["metaData"]["chip_params"]["Calib"].as<int>();
-        cond->moduleConfigs[0].injgain = 0;
-        if(yaml_file["metaData"]["chip_params"]["Inj_gain"])
-          cond->moduleConfigs[0].injgain = yaml_file["metaData"]["chip_params"]["Inj_gain"].as<int>();
-        
-        //for (const auto& params : config)
-        //  parseNode(params.first.as<std::string>(), params.second, cond);
+      // PARSE MAPPER
+      LogDebug("HGCalConfigESSourceFromYAML") << "Loading " << filename << "...";
+      const auto yaml_file = YAML::LoadFile(filename);
+      const auto mapper = yaml_file["ECONs"];
+      if (mapper.IsDefined()) {
+        for (const auto& params : mapper) { // loop through sequence of unnamed nodes
+          uint32_t id;
+          try {
+            id = params["id"].as<uint32_t>();
+          } catch (const YAML::ParserException& err) {
+            throw cms::Exception("HGCalConfigESSourceFromYAML") << "Bad conversion for id!" << err.msg;
+          }
+          std::string fname_ECOND = params["configs"]["ECOND"].as<std::string>();
+          std::string fname_ECONT = params["configs"]["ECONT"].as<std::string>();
+          //std::string fname_ROCs = params["configs"]["ROCs"].as<std::string>();
+          LogDebug("HGCalConfigESSourceFromYAML") << "Found module id=" << std::hex << id
+            << ", ECOND=" << fname_ECOND << ", ECONT=" << fname_ECONT;
+          //parseECONConfigYAML(fname_ECON,cond);
+          //parseROCConfigYAML(fname_ROCs,cond);
+          cond->moduleConfigs[0].gains[id+0*64] = 1;
+          cond->moduleConfigs[0].gains[id+1*64] = 1;
+          cond->moduleConfigs[0].gains[id+2*64] = 1;
+          cond->moduleConfigs[0].gains[id+3*64] = 1;
+          cond->moduleConfigs[0].gains[id+4*64] = 1;
+          cond->moduleConfigs[0].gains[id+5*64] = 1;
+        }
       } else {
         edm::LogWarning("HGCalConfigESSourceFromYAML")
-            << "The YAML configuration is missing a 'metaData' node. The conditions format may hence be invalid.\n"
+            << "The YAML configuration is missing a 'ECONs' node. The conditions format may hence be invalid.\n"
             << filename;
       }
       
-      //// PARSE ROC NODES
-      //for (const auto& params : yaml_file) { // loop through nodes
-      //  std::cout << "HGCalConfigESSourceFromYAML::parseYAML: Found key " << params.first.as<std::string>() << std::endl;
+      // PARSE META DATA NODE: place holders
+      cond->moduleConfigs[0].charMode = (bool) 0;
```

Comment:
Yes, I can maybe add this version parameter to manually set it in the python config until we implemented taking it from full YAML so it's easier to run `tb_raw2reco.py` without changing this hardcoded placeholder.

Reply:
@pfs, is it okay with manual overrides like this: https://github.com/CMS-HGCAL/cmssw/pull/31/commits/6648419545faa0229485297b542223e2ecc2257c
This would allow us to run the whole setup without relying on the current state of the YAML parser, as we haven't converged on a YAML format yet, let alone parsing it for relays?

Reply:
+1 (i imagine the stress)